src/gcode-parser/arc-gen_test.cc

```diff
@@ -154,6 +155,28 @@ static void testArcLength(GCodeParserAxis normal, bool clockwise, double start,
   }
 }
 
+TEST(ArcGenerator, ArcLength_NonEuclideanAxes) {
```

Comment:
To make more clear what we're testing, I'd probably name it
`ArcLength_InterpolateNonEuclidianAxes`.